pkg/cortex/modules.go

```diff
@@ -293,15 +290,27 @@ func (t *Cortex) initTableManager(cfg *Config) error {
 		os.Exit(1)
 	}
 
-	tableClient, err := storage.NewTableClient(lastConfig.IndexType, cfg.Storage)
-	if err != nil {
-		return err
+	var tableClients []chunk.TableClient
```

Comment:
Is there any reason we can't generate the table and bucket clients inside the NewTableManager function? Instead of instantiating the clients and passing them as a parameter would could just pass the storage config as a parameter. I think the way things are now is probably fine. However, the columnar relationship between the schema cfg and clients makes me think it might be better to encapsulate this logic inside the function. The comment I'm referring to:

```
// The i'th TableClient in 'tableClients' should correspond to the i'th PeriodConfig in schemaCfg.Config.
// The TableClient for the last client should not be nil.
func NewTableManager(cfg TableManagerConfig, schemaCfg SchemaConfig, maxChunkAge time.Duration, tableClients []TableClient, objectClient BucketClient) (*TableManager, error) {objectClient BucketClient) (*TableManager, error) {
```